Polyhedron/demo/Polyhedron/Scene_polyhedron_item.cpp

```diff
@@ -1497,7 +1497,9 @@ QString Scene_polyhedron_item::computeStats(int type)
     }
     else if(d->genus == -1)
     {
-      int s(d->poly->size_of_vertices()), a(d->poly->size_of_halfedges()/2), f(d->poly->size_of_facets());
+      int s(static_cast<int>(d->poly->size_of_vertices())),
+          a(static_cast<int>(d->poly->size_of_halfedges())/2),
+          f(static_cast<int>(d->poly->size_of_facets()));
       //cast in int to avoid intermediary state containing a negative size_t.
       d->genus = 1.0 - (s-a+f)/2.0;
```

Comment:
I do not understand the comment:
```C++
//cast in int to avoid intermediary state containing a negative size_t.
```

A `std::size_t` unsigned. If you want the "signed-`size_t`", then the type is `std::ptrdiff_t`, and not `int`!

I recommend to use the right types and less `static_cast` unless one has to use two different incompatible APIs.

... If you use `std::ptrdiff_t`, you will have to cast `s-a+f` to double:
```
d->genus = 1.0 - double(s-a+f)/2.0;
```
because the precision of `ptrdiff_t` might be 64 bits where a double only has 53 bits of precision in its mantissa... and compilers (like MSVC) will warn about a possible loss of precision.